kafka-ui-api/src/main/java/com/provectus/kafka/ui/jmx/JmxService.java

```diff
@@ -0,0 +1,34 @@
+package com.provectus.kafka.ui.jmx;
+
+import com.provectus.kafka.ui.cluster.model.KafkaCluster;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Service;
+
+import javax.management.MBeanServerConnection;
+import javax.management.ObjectName;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+import java.util.Map;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class JmxService {
+
+    @SneakyThrows
+    public void loadClusterMetrics(KafkaCluster kafkaCluster) {
+        String url = "service:jmx:rmi:///jndi/rmi://" + kafkaCluster.getJmxHost() + ":" + kafkaCluster.getJmxPort() + "/jmxrmi";
```

Comment:
Try to reuse connection, opening a new connection is an expensive operation.

Reply:
all connection are saved and reused now, fixed